warehouse/accounts/models.py

```diff
@@ -108,6 +108,15 @@ class User(SitemapMixin, db.Model):
         "UserEvent", backref="user", cascade="all, delete-orphan", lazy=True
     )
 
+    @property
+    def password(self):
+        return self._password
+
+    @password.setter
+    def password(self, value):
+        self._password = value
+        self.password_date = sql.func.now()
```

Comment:
I think we can commit the transaction early to get the new date and thus retain the existing DB triggers. I think it's probably desirable for those to continue to remain in the DB so we don't easily forget to update `password_date`.

Reply:
I can't figure out a way to get the resulting trigger value to be accessible inside the request. Calling `tm.commit()` mid request crashes the request at the end since txn complete... savepoint() doesn't work...

Reply:
trying sqlalchemy expire call... might work!

Reply:
... Ok, today is the day where my comments land 30s late.

Reply:
yah, absolutely not ideal, but this is no longer necessary with https://github.com/pypa/warehouse/pull/10859/commits/8309b0057b0677000a577de4d81facd39a23653a.

for clarity, sql.func.now() gets translated to `now()` on insert, it's not calling the function.

Reply:
(Yeah 👍  but to explicit my comment, if we had forced a commit at that point it would have been doing a SQL query in the setter)

Reply:
ah, I wasn't proposing a commit in the setter, I was saying I had attempted all of these things in the view.

Reply:
specifically, where refresh came to the rescue: https://github.com/pypa/warehouse/pull/10859/commits/8309b0057b0677000a577de4d81facd39a23653a#diff-55d42e8160b62924651b14b5c3649c6511c24a79876c8447cdad028da727d14fR364